### ISSUE TITLE
Add summary of GoSam's requirements and dependencies to README

### ISSUE THREAD

**Issue by matthewfeickert** [gudrunhe/gosam#10]
It is great to find GoSam on GitHub now, so thanks for moving it here. :+1: 

In the README it mentions

https://github.com/gudrunhe/gosam/blob/f47067670301d487b788a36e215e4c2b45902af2/README#L61

but to find all the dependencies one needs to dig a bit. Could you add a list of all of GoSam's requirements and dependencies to the README?

**Comment by Jens-Braun** [comment]
Dear @matthewfeickert, all dependencies are automatically installed by `tools/gosam_installer.py`, therefore versions of all requirements can be read from the installer (they seem to still be essentially the same as on the site you linked). The full list of the dependencies is:
- Python (≥3.6)
- Fortran Compiler (Any reasonably modern version of `gfortran` should work)
- QGRAF
- Form
- Dependencies packaged in [`gosam-contrib`](https://github.com/gudrunhe/gosam-contrib):
    - FF
    - QCDLoop
    - OneLOop
    - Golem95C
    - Samurai
    - Ninja
The complete installation procedure (including its documentation in the README) is completely revamped in the upcoming GoSam-3.0 release, which will also include a more exhaustive list of requirements.

**Comment by matthewfeickert** [comment]
> The full list of the dependencies is:
> 
> * Python (≥3.6)
> * Fortran Compiler (Any reasonably modern version of `gfortran` should work)
> * QGRAF
> * Form
> * Dependencies packaged in [`gosam-contrib`](https://github.com/gudrunhe/gosam-contrib):
>   
>   * FF
>   * QCDLoop
>   * OneLOop
>   * Golem95C
>   * Samurai
>   * Ninja

Thanks @Jens-Braun. As part of a [larger community project](https://github.com/hep-packaging-coordination/) we're looking at (the feasibility of) trying to package as much of HEP code as possible on [conda-forge](https://conda-forge.org/), and as conda-forge works via bootstrapping up dependencies from the compiler up (so all dependencies of a package need to be packaged on conda-forge before the dependent package can) it helps to have explicit manifests like this.

> The complete installation procedure (including its documentation in the README) is completely revamped in the upcoming GoSam-3.0 release, which will also include a more exhaustive list of requirements.

That's excellent to hear. :+1: Thanks for your work.

**Comment by gudrunhe** [comment]
Dear @matthewfeickert, the gosam-contrib will be obsolete with the upcoming release of GoSam-3.0, where we will get rid of some of the dependencies and package golem95C individually (including a license statement). Therefore we do not intend to maintain the gosam-contrib setup any longer.

**Comment by matthewfeickert** [comment]
Excellent to hear, @gudrunhe. :+1:

**Comment by matthewfeickert** [comment]
> We therefore wanted to ask whether you would still be interested in your original proposal of packaging GoSam for conda-forge?

@Jens-Braun It looks like as of 3ec55a29a30b7ec56dd4291e7d0b04723da66112 that your build system is now `meson`

https://github.com/gudrunhe/gosam/blob/3ec55a29a30b7ec56dd4291e7d0b04723da66112/Readme.md?plain=1#L42-L43

and your required dependencies are `QGRAF`, `Form`, and `Ninja`, with `Golem95` optional.

https://github.com/gudrunhe/gosam/blob/3ec55a29a30b7ec56dd4291e7d0b04723da66112/Readme.md?plain=1#L71-L74

Is that correct?

If so, only `Ninja` is on conda-forge (c.f. https://github.com/peraro/ninja?tab=readme-ov-file#install-from-conda-forge), so all the other dependencies need to go on conda-forge first.

* http://cfif.tecnico.ulisboa.pt/~paulo/qgraf.html isn't even loading, which is a bad sign in general. Where does this exist in a stable format?
* FORM exists on GitHub (https://github.com/form-dev/form) and so might be straightforward to package.
* Golem95 also exists on GitHub (https://github.com/gudrunhe/golem95) and is now suing Meson as well. What are Golem95's dependencies?

**Comment by Jens-Braun** [comment]
> [@Jens-Braun](https://github.com/Jens-Braun) It looks like as of [3ec55a2](https://github.com/gudrunhe/gosam/commit/3ec55a29a30b7ec56dd4291e7d0b04723da66112) that your build system is now `meson`

Yes, we are using `meson` for everything now. Direct dependencies are QGRAF, Form and Ninja now, Ninja requires OneLOop. Golem95 is optional and also requires OneLOop. 
Previous versions of GoSam had a `gosam_installer.py` script to install the dependencies, but now this is handled automatically through `meson`'s builtin `wrap` system. Therefore, from the outside perspective, GoSam is now a single package without dependencies. I know that this is against the philosophy of `conda-forge`, but it might be necessary since some patches are applied. Especially OneLOop needs to be patched in order to work properly with Golem95.

As for QGRAF: the link on the Readme was outdated, we updated it to the new one. Also the new page is not always reachable though. There is not really a way around this because of QGRAF's custom license, which forbids changes or redistribution. Since `conda-forge`  distributes build artifacts, it might be agaings the license conditions. So I am not sure if it would even be possible.

**Comment by matthewfeickert** [comment]
Thanks @Jens-Braun.

> Therefore, from the outside perspective, GoSam is now a single package without dependencies.

Can you elaborate? Meson doesn't remove the existence of dependencies, so I think I'm missing the point that you're making, sorry.

> I know that this is against the philosophy of conda-forge, but it might be necessary since some patches are applied. Especially OneLOop needs to be patched in order to work properly with Golem95.

What are the patches? Have they been upstreamed, or at least discussed with the dependency authors? [Patches](https://rattler.build/latest/reference/recipe_file/#patches) are common on conda-forge, so that wouldn't be an issue, but the assumption is that if you're patching something there's a good reason for it and that (hopefully) the patch will be not needed in future versions. I already have oneLOop on conda-forge (https://github.com/conda-forge/oneloop-feedstock) so it would be interesting to understand what's required, and if [build variants](https://rattler.build/latest/variants/) would resolve what is needed for Golem95.

> There is not really a way around this because of QGRAF's custom license, which forbids changes or redistribution.

🤯 I have never seen that in scientific software before &mdash; interesting. But yeah, assuming that http://cefema-gt.tecnico.ulisboa.pt/~paulo/ld.html is  correct and has up to date information,

> The Programs are copyrighted, and are not to be redistributed without explicit permission — in whatever form — except in very specific circumstances, if any, described in the header of the corresponding Fortran file(s).

is pretty clear on that. Do you or anyone you work with know if this (binary redistribution) has ever been discussed with the author? Correct me if I'm wrong, but I don't think this has been a common thing for Fortran in the past and so I could see that this might not have been something considered too deeply if the license isn't newly revised.

Without a change though, it would be "game over" on getting anything that depends on QGRAF on conda-forge.
